src/shogun/lib/type_case.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#ifndef SHOGUN_TYPE_CASE_H
+#define SHOGUN_TYPE_CASE_H
+
+#include <typeindex>
+#include <unordered_map>
+
+#include <shogun/lib/any.h>
+#include <shogun/lib/type_list.h>
+
+using namespace shogun;
+
+// here we define a list of all types that are implemented in shogun
+// the head of the list returns the type
+typedef Types<
+    bool, char, int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t, int64_t,
+    uint64_t, float32_t, float64_t, floatmax_t>::type AllTypes;
+
+namespace shogun
+{
+	// here we make types switchable
+	enum class TYPE
+	{
+		PT_BOOL = 1,
```

Comment:
the PT_ prefix comes from somewhere else?

Reply:
yes, that comes from `shogun/tests/unit/base/SGObjectAll_unittest.cc`

Reply:
i see, what about somehow re-using those from `DataType.h`

Reply:
Ah seem to be the same! But will have to address the containers..

Reply:
Well we don't want to support *all* the containers. Also the container definitions are part of the old type system and probably will be removed soon (nobody needs strings of SGVectors).
What is more likely is that we support matrix/vector of the ptypes, and then strings...but let's do that later when it is needed.